skbio/diversity/_driver.py

```diff
@@ -180,9 +182,114 @@ def alpha_diversity(metric, counts, ids=None, validate=True, **kwargs):
     return pd.Series(results, index=ids)
 
 
+@deprecated(as_of='0.4.2-dev', until='0.5.0',
+            reason=('The return type is unstable. Developer caution is '
+                    'advised. The resulting DistanceMatrix object may include '
+                    'zeros when distance has not been calculated, and '
+                    'therefore can be misleading.'))
+def partial_beta_diversity(metric, counts, ids=None, validate=None,
```

Comment:
@gregcaporaso and I don't think the signatures need to be the same since `partial_beta_diversity` is deprecated and slated for removal/change. A user shouldn't inadvertently use `partial_beta_diversity` where they meant to be using `beta_diversity`. Also, it doesn't make sense to adapt the `beta_diversity` `experimental` API to match the `deprecated` `partial_beta_diversity` API.